packages/connect-node/src/grpc-http-transport.ts

```diff
@@ -437,9 +433,25 @@ function makeNodeRequest(options: NodeRequestOptions) {
   });
 }
 
-function nodeRequest(protocol: string) {
-  if (protocol.startsWith("http") || protocol.startsWith("https")) {
-    return protocol.includes("https") ? https.request : http.request;
-  }
-  throw new Error("Unsupported protocol");
+function getNodeRequest(
```

Comment:
I think it's fair to deduplicate this function now, and move it to `src/private/node-request.ts` for example 🙂

Reply:
moved all node request functions to `node-request.ts` in private and replaced all duplicated code